File: backend/src/main/java/com/hercare/backend/service/AppointmentService.java
package com.hercare.backend.service;

import java.util.List;

import com.hercare.backend.dto.request.AppointmentRequest;
import com.hercare.backend.dto.response.AppointmentResponse;

public interface AppointmentService {

    AppointmentResponse bookAppointment(
            AppointmentRequest request);

    List<AppointmentResponse> getMyAppointments();

    List<AppointmentResponse> getDoctorAppointments();

    AppointmentResponse approveAppointment(Long id);

    AppointmentResponse rejectAppointment(Long id);

    AppointmentResponse cancelAppointment(Long id);

    AppointmentResponse completeAppointment(Long id);
}

File: frontend/src/services/appointmentService.js
import axiosClient from '../api/axiosClient'

const appointmentService = {
  bookAppointment: (payload) => axiosClient.post('/appointments', payload),
  getMyAppointments: () => axiosClient.get('/appointments/my'),
  getDoctorAppointments: () => axiosClient.get('/appointments/doctor'),
  approveAppointment: (id) => axiosClient.put(`/appointments/${id}/approve`),
  rejectAppointment: (id) => axiosClient.put(`/appointments/${id}/reject`),
  completeAppointment: (id) => axiosClient.put(`/appointments/${id}/complete`),
  cancelAppointment: (id) => axiosClient.put(`/appointments/${id}/cancel`),
}

export default appointmentService
